### PR TITLE
LMN-5543 | Subscribe to HealthKit changes

### PR DESCRIPTION
**Pull request by a-vachasov**
- created workouts observer to receive update triggers from health kit.

### CODE REVIEW

File: ios/Classes/SwiftFlutterHealthFitPlugin.swift [hunk 1]
```diff
@@ -54,7 +57,11 @@ public class SwiftFlutterHealthFitPlugin: NSObject, FlutterPlugin {
             return
         }
         
-        guard UIApplication.shared.applicationState == .active else {
+        // TODO: (a-vachasov): In order to sync-up workouts in background we need to bypass this check.
+        // TODO: (a-vachasov): Why do we even need this check?
+        guard UIApplication.shared.applicationState == .active
+                || call.method == "isWorkoutsAuthorized"
+                || call.method == "getWorkoutsBySegment" else {
```
Comment (a-vachasov) [inline ios/Classes/SwiftFlutterHealthFitPlugin.swift:64]: @NimGit @rtokun Guys, do you have any ideas why do we need it?

I need to bypass it to update workouts in background.

Comment (NimGit) [reply ios/Classes/SwiftFlutterHealthFitPlugin.swift:64]: the error in the guard reads 
result(FlutterError(code: "background call", message: "cannot read from healthkit on background", details:""))

I think this is the reason

Comment (a-vachasov) [reply ios/Classes/SwiftFlutterHealthFitPlugin.swift]: I tested it with background mode, flutter can't do any calls to ios native platform in background.
Background state doesn't mean the app is not on the screen, it's a special state of the app when it's awaken by the system to do background processing.

File: ios/Classes/WorkoutsObserver.swift [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    var observerQuery: HKObserverQuery?
+    
+    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
+        eventSink = events
+        self.startObservation()
+        return nil
+    }
+    
+    private func startObservation() {
+        print("WorkoutsObserver started observation")
```
Comment (NimGit) [inline ios/Classes/WorkoutsObserver.swift:16]: do we need this print?

**Review by NimGit — APPROVED** [review]

**Comment by VusDanylo** [comment]
I think we also need ability to stop subscription. For example after log out.

File: ios/Classes/WorkoutsObserver.swift (hunk 2 above)
Comment (a-vachasov) [reply ios/Classes/WorkoutsObserver.swift:16]: It's not mandatory for sure, just added it for visibility

**Pull request merged**